### PR TITLE
Global Settings implementation

### LINKED ISSUE #89: Global Settings (Change Colour Theme and Log Out)
Settings list in the top right toolbar. 

Breakdown of designs and functionality:
Screen Shot 2019-07-24 at 11 20 57 pm
1. User clicks top right profile icon, and it drops a number of options (similar to this animation: https://codepen.io/balapa/pen/zvObzO -  but not as dramatic and not with expanding tiles - gentle dropdown of options one after another but at a quickened pace so it doesnt take too long )

2. Options hovered over come to the foreground to demonstrate it is potentially selected (ie more shadowing behind the tile)

3. Log out button is moved to these options, and keeps the same functionality - old log out button is removed

Screen Shot 2019-07-24 at 11 21 04 pm
1. When a tile is clicked it moves to the top to demonstrate it is selected, and it drops down sub-option(s)

2. X tile closes the options list and resets the top tab to the profile icon. Additionally, clicking on the top tab above the divider will retract all options and reset to profile icon.

3. Back arrow moves user back to previous options 

Screen Shot 2019-07-24 at 11 21 10 pm
1. User picks a colour, which changes the application theme to one of the other colours that are selectable at the beginning when you first install the application (should show all options except the colour theme already selected) 

Additional info: you may have to resize/reposition some of the elements (except the 'Speckle' logo) in the toolbar to fit the button in

**NOTE: Please keep this well-documented because we will need to add more options, settings etc down the track. If we can keep this modular and easily upgraded/maintained that would be preferable.**

### PR DESCRIPTION
**Pull request by mul1sh**
Fixes #89

### CODE REVIEW

**Comment by fyang1024** [comment]
Hi @mul1sh , we have reviewed the changes and happy with your work. One more thing, can you please solve conflicts in package.json. I believe they are just lib versions differences.

File: src/ts/components/dashboard/SettingsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
+import * as React from 'react'
+import { TweenMax, TimelineLite } from 'gsap/all'
```
Comment (mytechtip) [inline src/ts/components/dashboard/SettingsMenu.tsx]: Please remove `TimelineLite` if not used

**Comment by mul1sh** [comment]
@mytechtip @fyang1024  thanks for the review, I've resolved the conflicts and removed the unused imports as required :)

**Comment by anjj123** [comment]
Thanks @mul1sh, this looks fantastic! Can you make a few small edits to the UI? 

1. When hovering your cursor over the button, it remains as a static pointer and is not reactive;
2. Some minor design issues:
2.1. The settings button should always have the rectangular border around it, even when not selected (atm, it changes between both when clicked);
2.2. Can we make the chain switching bar smaller in width or move it a bit more to the left, so we can center it and leave more room for the 'Speckle' logo in the left? Looks too crammed

**Comment by mul1sh** [comment]
@anjj123  sure no worries, doing these changes now 🙂

**Pull request merged**